gutenTAG/base_oscillations/random_walk.py

```diff
@@ -30,28 +34,16 @@ def generate_only_base(self,
         amplitude = amplitude or self.amplitude
         smoothing = smoothing or self.smoothing
 
-        return random_walk(ctx.rng, length, amplitude, smoothing)
-
-
-def _gen_steps(rng: np.random.Generator, length: int) -> np.ndarray:
-    steps = rng.choice([-1., 0., 1.], size=length - 1)
-    return np.r_[0, steps].cumsum()
-
+        if smoothing:
+            filter_size = int(smoothing * length)
+            ts = _gen_steps(ctx, length + filter_size - 1)
+            gaussian = norm.pdf(np.linspace(-1.5, 1.5, filter_size))
+            ts_filter = gaussian / gaussian.sum()
+            ts = np.convolve(ts, ts_filter, "valid")
+        else:
+            ts = _gen_steps(ctx, length)
 
-def random_walk(rng: np.random.Generator = np.random.default_rng(),
```

Comment:
related to the question regarding the import: Removing this method does not impose any side effects, e.g. the method is not needed somewhere else?

Reply:
Also, here I don't recall removing it. @CodeLionX should maybe answer it

Reply:
This seems part of the same refactoring as the comment above

Reply:
See above. Thanks for having a look.

Reply:
This file is still in the diff.